dockstore-client/src/main/java/io/dockstore/client/cli/nested/WorkflowClient.java

```diff
@@ -663,9 +664,29 @@ protected void handlePublishUnpublish(String entryPath, String newName, boolean
                     publish(true, entryPath);
                 }
             } else {
-                //for workflows method currently doesn't work with --entryname flag
-                errorMessage("Parameter '--entryname' not valid for workflows. See `workflow publish --help` for more information.",
-                    CLIENT_ERROR);
+                try {
```

Comment:
This was also like this, but for the block starting on line 652/653, I feel that we should have a message saying `newName` is not valid for unpublish.

Reply:
The current implementation doesn't handle publishing/unpublishing workflows with '--entryname' gracefully if there is an error. For example, if I try to publish a workflow with a custom name that is already published, the function will throw an ApiException instead of printing a clean "This workflow is already published" message.

I'll fix this in my next commit, but would like to clarify expected functionality:
`dockstore workflow publish --unpub --entry workflowpath --entryname newname` 
vs 
`dockstore workflow publish --unpub --entry workflowpath/entryname`

Currently, in the first case, the entry name is ignored and the command is interpreted as: 
`dockstore workflow publish --unpub --entry workflowpath`.
 Would it make sense to have the two cases listed above be logically equivalent in execution? Or should the function prevent use of --unpub with --entryname?

Reply:
I will defer to @garyluu and @denis-yuen , but here are my thoughts:

1. If we're going to support `--entryname` we should add it to the [help message](https://github.com/dockstore/dockstore-cli/blob/5de67a497465612ddc693ffc694e916eea018af2/dockstore-client/src/main/java/io/dockstore/client/cli/nested/WorkflowClient.java#L689). I didn't realize until reading the issue again that this an enhancement.
2. Depending on how you want to approach it, either that help message should say that the argument is ignored for unpublish; or, the help message says it's only applicable to publish and if we detect it here for unpublish, we throw an error.

In either case, I think this renders your question moot; --entryname should either be ignored or cause an error if used with unpublish.

In hindsight, I wished we had called it something like `newEntryName`.

Reply:
I'm personally not a fan of flags that conflict with other flags when it can be helped. I'd rather there be an unpublish command. However, in order to keep things similar to tools, I suppose that shouldn't be done.

Anyways, given the current situation, I think I agree with Charles. What does tools do right now?

Reply:
Tools functions in a [similar way](https://github.com/dockstore/dockstore-cli/blob/5de67a497465612ddc693ffc694e916eea018af2/dockstore-client/src/main/java/io/dockstore/client/cli/nested/ToolClient.java#L213) to workflows. The `--entryname` parameter is ignored for unpublish, regardless of if it is set or not. Both could use slightly more descriptive error messages for when publish/unpublish fails.

I can modify workflows to better handle `--entryname` with unpublish and provide more descriptive errors. Should i do the same for tools/create a similar ticket for tools? Furthermore, I think @coverbeck's idea to change `--entryname` to something such as `--newEntryName` would be beneficial.

Reply:
The problem is, even though it's extremely unlikely, is that there could be some script somewhere using `--entryname`, and changing the parameter name would break that script. Some ideas:

1. Only make it `newEntryName` for workflows, since it's a new feature there. The downside is that it's asymmetrical with tools though.
2. Make `--newEntryName` an additional option for tools. Continue to support `--entryname`, but undocumented, i.e., the help message for tools only lists `--newEntryName`.
3. It's unlikely anybody is using this feature, just change it on tools.

I suggest the first option for now, and wait for Denis' opinion.

Reply:
I think if you have time to make all the changes (for both tools and workflows) mentioned by Charles then go ahead. Otherwise, I think you should first match the functionality of tools and workflows.   And then create tickets to simultaneously make improvements to both.

Reply:
I'll update both tools and workflows to have the same handling of the `--entryname` parameter, to ensure they are logically symmetric. I'll move forward from there after getting Denis' opinion.

Reply:
This looks like its going into develop ... I'm ok with 2. actually (rename to better parameter name but allow old name for backwards compatibility). Also ok with a series of PRs for the same reason (I've be more nervous about a partial incomplete fix for 1.9.x)

Reply:
Ok, I'll update `--entryName` to `--newEntryName`, and modify the help messages to specify `--newEntryName` but still keep support for `--entryName`.